jwt/claims.go

```diff
@@ -0,0 +1,90 @@
+/*-
+ * Copyright 2014 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jwt
+
+import (
+	"time"
+
+	"github.com/square/go-jose"
+)
+
+// Claims represents public claim values defined in RFC7519
+type Claims struct {
+	Issuer    string    `json:"-"`
+	Subject   string    `json:"-"`
+	Audience  []string  `json:"-"`
+	Expiry    time.Time `json:"-"`
+	NotBefore time.Time `json:"-"`
+	IssuedAt  time.Time `json:"-"`
+	ID        string    `json:"-"`
+}
+
+type rawClaims struct {
+	Iss string      `json:"iss,omitempty"`
+	Sub string      `json:"sub,omitempty"`
+	Aud audience    `json:"aud,omitempty"`
+	Exp NumericDate `json:"exp,omitempty"`
+	Nbf NumericDate `json:"nbf,omitempty"`
+	Iat NumericDate `json:"iat,omitempty"`
+	Jti string      `json:"jti,omitempty"`
+}
+
+func (c *Claims) marshalJSON() ([]byte, error) {
```

Comment:
You can make this `MarshalJSON` and `UnmarshalJSON` below to make it directly serializeable/deserializeable.